src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -0,0 +1,388 @@
+            {
+                _version++;
+                _enumStoreBuilder = value;
+            }
+        }
+
+        internal IEnumerable<TexlFunction> Functions => _functions;
+
+        IExternalDocument INameResolver.Document => _document;
+
+        IExternalEntityScope INameResolver.EntityScope => throw new NotImplementedException();
+
+        IExternalEntity INameResolver.CurrentEntity => default;
+
+        DName INameResolver.CurrentProperty => default;
+
+        DPath INameResolver.CurrentEntityPath => default;
+
+        bool INameResolver.SuggestUnqualifiedEnums => false;
+
+        IEnumerable<TexlFunction> INameResolver.Functions => _functions;
+
+        internal readonly Dictionary<DName, IExternalEntity> _environmentSymbols = new Dictionary<DName, IExternalEntity>();
+
+        internal DisplayNameProvider _environmentSymbolDisplayNameProvider = new SingleSourceDisplayNameProvider();
+                
+        IReadOnlyDictionary<string, NameLookupInfo> IGlobalSymbolNameResolver.GlobalSymbols => _variables;
+        
+        /// <summary>
+        /// Provide variable for binding only.
+        /// Value must be provided at runtime via <see cref="RuntimeConfig.SymbolValues"/>.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="type"></param>
+        public void AddVariable(string name, FormulaType type)
+        {
+            _version++;
+
+            // $$$ does it Overwrite existing?
+            _variables[name] = new NameLookupInfo(
+                BindKind.PowerFxResolvedObject,
+                type._type,
+                DPath.Root,
+                0,
+                data: new NameSymbol(name));
+        }
+
+        /// <summary>
+        /// Provide variable for binding, with a runtime value. 
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="type"></param>
+        /// <param name="data"></param>
+        public void AddVariable(string name, FormulaType type, ICanGetValue data)
+        {
+            _version++;
+            
+            // $$$ does it Overwrite existing?
+            _variables[name] = new NameLookupInfo(
+                BindKind.PowerFxResolvedObject, 
+                type._type, 
+                DPath.Root, 
+                0, 
+                data);
+        }
+
+        /// <summary>
+        /// Remove variable of given name. 
+        /// </summary>
+        /// <param name="name"></param>
+        public void RemoveVariable(string name)
+        {
+            _version++;
+
+            // $$$ Succeed if missing
+            _variables.Remove(name);
+        }
+
+        /// <summary>
+        /// Remove function of given name. 
+        /// </summary>
+        /// <param name="name"></param>
+        public void RemoveFunction(string name)
+        {
+            _version++;
+
+            // $$$ namespace?
+            _functions.RemoveAll(func => func.Name == name);
+        }
+
```

Comment:
We probably need RemoveFunction(func) as well to remove an overload and not all funcitons with the same name